core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -215,12 +213,24 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         })
       }
 
-      lastModifiedTime = newLastModifiedTime
+      lastScanTime = newLastScanTime
     } catch {
       case e: Exception => logError("Exception in checking for event log updates", e)
     }
   }
 
+  private def getNewLastScanTime(): Long = {
+    val fileName = "." + UUID.randomUUID().toString
+    val path = new Path(logDir, fileName)
+    val fos = fs.create(path)
+    fos.close()
+
+    val newLastScanTime = fs.getFileStatus(path).getModificationTime
+    fs.delete(path)
```

Comment:
You need this line to be in a `finally` block after the `fs.create()` call. Something like:

    val fileName = "." + UUID.randomUUID().toString
    val path = new Path(logDir, fileName)
    val fos = fs.create(path)
    val newLastScanTime =
      try {
        fos.close()
        fs.getFileStatus(path).getModificationTime
      } finally {
        fs.delete(path)
      }